### PR TITLE
UnixResolverDnsServerAddressStreamProvider default name server selection and ordering bug

### LINKED ISSUE #6736: Incorrect UnixResolverDnsServerAddressStreamProvider
### Minimal yet complete reproducer code (or URL to code)

```java
import static org.junit.Assert.assertNotNull;

import java.io.File;
import java.io.FileOutputStream;
import java.io.OutputStream;
import java.nio.charset.StandardCharsets;

import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.TemporaryFolder;

import io.netty.resolver.dns.UnixResolverDnsServerAddressStreamProvider;

public class ResolverTest {

    @Rule
    public final TemporaryFolder folder = new TemporaryFolder();

    @Test
    public void test() throws Exception {
        File f = folder.newFile();
        try (OutputStream out = new FileOutputStream(f)) {
            out.write(("domain linecorp.local\n" +
                       "nameserver 1.2.3.4\n" +
                       "nameserver 5.6.7.8\n").getBytes(StandardCharsets.UTF_8));
        }
        UnixResolverDnsServerAddressStreamProvider p =
                new UnixResolverDnsServerAddressStreamProvider(f, null);

        assertNotNull(p.nameServerAddressStream("somehost"));
    }
}
```

### Expected behavior

`UnixResolverDnsServerAddressStreamProvider.nameServerAddressStream()` should not return `null`.

The expected behavior of `DnsNameResolver` is to try `somehost.` and then `somehost.linecorp.local.`

### Actual behavior

Attempting to resolve a host will trigger a NPE:

```
java.lang.NullPointerException: nameServerAddrs
 at io.netty.util.internal.ObjectUtil.checkNotNull(ObjectUtil.java:31)
 at io.netty.resolver.dns.DnsNameResolverContext.<init>(DnsNameResolverContext.java:120)
 at io.netty.resolver.dns.DnsNameResolver$SingleResolverContext.<init>(DnsNameResolver.java:659)
 at io.netty.resolver.dns.DnsNameResolver.doResolveUncached(DnsNameResolver.java:653)
 at io.netty.resolver.dns.DnsNameResolver.doResolve(DnsNameResolver.java:590)
 at io.netty.resolver.dns.DnsNameResolver.doResolve(DnsNameResolver.java:523)
 at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:63)
 at io.netty.resolver.dns.InflightNameResolver.resolve(InflightNameResolver.java:100)
 at io.netty.resolver.dns.InflightNameResolver.resolve(InflightNameResolver.java:66)
 at io.netty.resolver.dns.InflightNameResolver.resolve(InflightNameResolver.java:51)
 at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:57)
 at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:32)
 at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:108)
 at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:208)
 at io.netty.bootstrap.Bootstrap.doResolveAndConnect(Bootstrap.java:170)
 at io.netty.bootstrap.Bootstrap.connect(Bootstrap.java:145)
```
### Netty version

4.1.11

### Workaround

Use `DefaultDnsServerAddressStreamProvider.INSTANCE` instead of `DnsServerAddressStreamProviders.platformDefault()`.

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
UnixResolverDnsServerAddressStreamProvider allows the default name server address stream to be null, but there should always be a default stream to fall back to ([1] Search Strategy).
UnixResolverDnsServerAddressStreamProvider currently shuffles the names servers are multiple are present, but the defined behavior is to try them sequentially [2].

[1] Search Strategy Section - https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man5/resolver.5.html
[2] DESCRIPTION/nameserver Section - https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man5/resolver.5.html

Modifications:
- UnixResolverDnsServerAddressStreamProvider should always use the first file provided to derive the default domain server address stream. Currently if there are multiple domain names in the file identified by the first argument of the constructor then one will be selected at random.
- UnixResolverDnsServerAddressStreamProvider should return name servers sequentially.
- Reduce access level on some methods which don't have known use-cases externally.

Result:
Fixes #6736

### CODE REVIEW

File: resolver-dns/src/main/java/io/netty/resolver/dns/UnixResolverDnsServerAddressStreamProvider.java [hunk 1]
```diff
@@ -217,7 +222,7 @@ private static void putIfAbsent(Map<String, DnsServerAddresses> domainToNameServ
                                     String domainName,
                                     List<InetSocketAddress> addresses) {
         // TODO(scott): sortlist is being ignored.
-        putIfAbsent(domainToNameServerStreamMap, domainName, DnsServerAddresses.shuffled(addresses));
+        putIfAbsent(domainToNameServerStreamMap, domainName, DnsServerAddresses.sequential(addresses));
```
Comment (normanmaurer) [inline resolver-dns/src/main/java/io/netty/resolver/dns/UnixResolverDnsServerAddressStreamProvider.java]: this seems to be not related to the PR. can you revert ?

Comment (Scottmitch) [inline resolver-dns/src/main/java/io/netty/resolver/dns/UnixResolverDnsServerAddressStreamProvider.java]: Yes this is not directly related to this change but it makes unit testing more difficult. If I pull this out I would also pull the unit tests out (and submit them together) ... is that preferred from your approach?

https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man5/resolver.5.html
```
If there are multiple servers,
     the resolver library queries them in the order listed.  The algorithm used is to try a name server, and
     if the query times out, try the next, until out of name servers, then repeat trying all the name
     servers until a maximum number of retries are made.
```

Comment (normanmaurer) [inline resolver-dns/src/main/java/io/netty/resolver/dns/UnixResolverDnsServerAddressStreamProvider.java]: its fine then... Just was not sure why you did this.

Comment (Scottmitch) [inline resolver-dns/src/main/java/io/netty/resolver/dns/UnixResolverDnsServerAddressStreamProvider.java]: I will update the commit message

**Comment by Scottmitch** [comment]
4.1 (0f1a2ca5a)
